### PR TITLE
Update hpxml2hescore duct validation

### PR DESCRIPTION
**Pull request by bpark1327**
Addresses #296

## Pull Request Description

Make sure heating and cooling systems that are not allowed to have a distribution system don't have them.

## Checklist

PR Author: Check these when they're done. Not all may apply. ~~strikethrough~~ and check any that do not apply. 

PR Reviewer: Verify each has been completed.

- [ ] Code changes (must work)
- [ ] Test exercising your feature or bug fix. Check the coverage report in the build artifacts.
- [ ] All other unit tests passing
- [ ] Update translation docs

### CODE REVIEW

File: hescorehpxml/schemas/hescore_json.schema.json [hunk 1]
```diff
@@ -1654,17 +1654,6 @@
...
-                                                    "properties": {
-                                                        "heating": {
-                                                            "properties": {
-                                                                "type": {"enum": ["wall_furnace", "baseboard", "boiler", "mini_split", "wood_stove", "none"]}
-                                                            },
-                                                            "required": ["type"]
-                                                        }
-                                                    },
-                                                    "required": ["heating"]
-                                                },
```
Comment (nmerket) [inline hescorehpxml/schemas/hescore_json.schema.json:1667]: Did you intend to remove this?

File: hescorehpxml/base.py [hunk 2]
```diff
@@ -1979,17 +1979,32 @@ def get_duct_mapping(element_list):
...
-                                       htg_sys_id)
+                raise TranslationError(
+                    f'Heating system {htg_sys_id} is not associated with an air distribution system.')
+            try:
+                htg_sys_dist_id = [key if value == htg_sys_id else None for key, value in dist_heating_map.items()][0]
+            except IndexError:
+                continue
+            if htg_sys['type'] not in heating_sys_types_requiring_ducts and htg_sys_dist_id is not None and\
+                    distribution_systems[htg_sys_dist_id] is not None:
+                raise TranslationError(f'Ducts are not allowed for heating system {htg_sys_id}.')
```
Comment (nmerket) [inline hescorehpxml/base.py:1995]: This branch isn't hit by a test. Can you add one that ensures that this is working properly?

File: hescorehpxml/base.py [hunk 3]
```diff
@@ -1979,17 +1979,32 @@ def get_duct_mapping(element_list):
...
-                                       clg_sys_id)
+                raise TranslationError(
+                    f'Cooling system {clg_sys_id} is not associated with an air distribution system.')
+            try:
+                clg_sys_dist_id = [key if value == clg_sys_id else None for key, value in dist_cooling_map.items()][0]
+            except IndexError:
+                continue
+            if clg_sys['type'] not in cooling_sys_types_requiring_ducts and clg_sys_dist_id is not None and\
+                    distribution_systems[clg_sys_dist_id] is not None:
+                raise TranslationError(f'Ducts are not allowed for cooling system {clg_sys_id}.')
```
Comment (nmerket) [inline hescorehpxml/base.py:2007]: This branch isn't hit by a test. Can you add one that ensures that this is working properly?

File: hescorehpxml/schemas/hescore_json.schema.json (hunk 1 above)
Comment (bpark1327) [reply hescorehpxml/schemas/hescore_json.schema.json:1667]: Yes, I intended to remove it. It is unnecessary.

**Review by nmerket — APPROVED** [review]

**Pull request merged**